hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,329 @@
+
+        # Reconstructing reference wave and applying it (division):
+        if reference is None:
+            wave_reference = 1
+        elif reference.axes_manager.navigation_size != self.axes_manager.navigation_size:  # case when reference is 1d
+            wave_reference = reference.deepcopy()
+
+            # Prepare parameters for reconstruction of the reference wave:
+
+            if reference.axes_manager.navigation_size != sb_position.axes_manager.navigation_size:  # 1d reference, but
+                # parameters are multidimensional
+                sb_position_ref = sb_position.inav[0].data
+            else:
+                sb_position_ref = sb_position_temp
+
+            if reference.axes_manager.navigation_size != sb_size.axes_manager.navigation_size:  # 1d reference, but
+                # parameters are multidimensional
+                sb_size_ref = np.float64(sb_size.inav[0].data)
+            else:
+                sb_size_ref = sb_size_temp
+
+            if reference.axes_manager.navigation_size != sb_smoothness.axes_manager.navigation_size:  # 1d reference, but
+                # parameters are multidimensional
+                sb_smoothness_ref = np.float64(sb_smoothness.inav[0].data)
+            else:
+                sb_smoothness_ref = sb_smoothness_temp
+            #
+
+            wave_reference.map(reconstruct, holo_sampling=self.sampling, sb_size=sb_size_ref,
+                               sb_position=sb_position_ref, sb_smoothness=sb_smoothness_ref, output_shape=output_shape,
+                               plotting=plotting, show_progressbar=show_progressbar)
+
+        else:
+            wave_reference = reference.deepcopy()
+            wave_reference.map(reconstruct, holo_sampling=self.sampling, sb_size=sb_size_temp,
+                               sb_position=sb_position_temp, sb_smoothness=sb_smoothness_temp,
+                               output_shape=output_shape, plotting=plotting, show_progressbar=show_progressbar)
+
+        wave_image = wave_object / wave_reference
+
+        wave_image.set_signal_type('electron_wave')  # New signal is a wave image!
+
+        wave_image.axes_manager.signal_axes[0].scale = self.sampling[0] * self.axes_manager.signal_shape[0] / \
+                                                       output_shape[0]
+        wave_image.axes_manager.signal_axes[1].scale = self.sampling[1] * self.axes_manager.signal_shape[1] / \
+                                                       output_shape[1]
+
+        # Reconstruction parameters are stored in holo_reconstruction_parameters:
+        rec_param_dict = OrderedDict([('sb_position', sb_position_temp), ('sb_size', sb_size_temp),
+                                      ('sb_units', sb_unit), ('sb_smoothness',
+                                                              sb_smoothness_temp)])
+
+        wave_image.metadata.Signal.add_node('holo_reconstruction_parameters')
```

Comment:
The intention here is that we like sometimes to check the feasibility of the parameters and to often to compare those across the datasets or different analyses trials. This actually needed as we most of the time use reconstruction method without providing parameters (the automatic routines seem to be very reliable). So will do both: make storing the parameters optional and inside node `.Holography`. In future defaults can be stored in global default of HS to provide users a possibility to enable or disable this feature for all analyses.